doc/flame/collision_detection.md

```diff
@@ -223,6 +223,135 @@ class MyGame extends FlameGame with HasCollisionDetection {
 ```
 
 
+## Ray casting and Ray tracing
+
+Ray casting and ray tracing are methods of sending out rays from a point in space in your game and
+be able to see what the rays collide with and how their reflections would be after hitting
+something.
+
+
+### Ray casting
+
+Ray casting is the operation of casting out one or more rays from a point and see if they hit
+anything, in Flame's case, hitboxes.
+
+We provide two methods for doing so, `raycast` and `raycastAll`. The first one just casts out
+a single ray and gets back a result with information about what and where the ray hit, and some
+extra information like the distance, normal and the reflection ray. The second one, `raycastAll`,
+works similarly but sends out multiple rays uniformly all around the origin, or within two angles
+from the origin.
+
+To use the ray casting functionality you have to have the `HasCollisionDetection` mixin on your
+game. After you have added that you can call `collisionDetection.raycast(...)` on your game class.
+
+Example:
+
+```dart
+class MyGame extends FlameGame with HasCollisionDetection {
+  @override
+  void update(double dt) {
+    super.update(dt);
+    final ray = Ray2(Vector2(0, 100), Vector2(1, 0));
```

Comment:
Is it too late to change the signature of `Ray2` to have the arguments named? I.e. `Ray2(origin: Vector2(0, 100), direction: Vector2(1, 0))`. Otherwise, it may be confusing as to which vector represents what.

Reply:
I had it like that first, but I heard it is a bit more expensive with named arguments, but I have no idea of how much more expensive. So maybe it is worth it?

Reply:
Ok, let's keep as-is then.

Reply:
Haha, just updated! It is so much easier to read.